### PR TITLE
More fixes and refactoring for Couch to SQL forms and cases migration

### PR DESCRIPTION
**Pull request by millerdev**
Review by commit 🐠 

Note on https://github.com/dimagi/commcare-hq/commit/69cbd41ee51c5f9c2ac21ed410366ed87cd3a68b: unfortunately I was unable to come up with a test that reproduced the issue, but it did allow a domain to be migrated that was previously failing.

@snopoke @mkangia @czue

### CODE REVIEW

**Review by czue — APPROVED** [review]
looks good to me

File: corehq/tests/noseplugins/logfile.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def setup_log(self):
     def log(self, label, test, err):
         if self.log_file is None:
             self.setup_log()
+        if isinstance(err[1], six.text_type):
+            value = type(err[0].__name__, (Exception,), {})(err[1])
+            err = (err[0], value, err[2])
```
Comment (mkangia) [inline corehq/tests/noseplugins/logfile.py]: nit: can you add a note on whats happening here?

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 2]
```diff
@@ -816,6 +817,12 @@ def _get_case_and_ledger_updates(domain, sql_form):
     )
 
 
+def _get_case_ids(xform):
+    case_ids = get_case_ids_from_form(xform)
+    case_ids.update(get_case_ids_from_stock_transactions(xform))
```
Comment (mkangia) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: how is this resolving the duplicate key issue?

**Review by mkangia — APPROVED** [review]
Don't have context on the changes but review from the code perspective.

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: I think the issue was that the migration was attempting to process 2 forms in simultaneously which is what was causing the duplicate key issue. This change adds case IDs referenced by stock transactions into the list of IDs what are used to determine if it's safe to process a form (if no other form already being processed touches any of these IDs)

Comment (snopoke) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: Looking at the other places where `get_case_ids_from_form` is called, most also call `get_case_ids_from_stock_transactions` and the places that don't likely should. Perhaps worth making a function that combines the 2?

**Review by snopoke — APPROVED** [review]

File: corehq/tests/noseplugins/logfile.py (hunk 1 above)
Comment (millerdev) [reply corehq/tests/noseplugins/logfile.py]: https://github.com/dimagi/commcare-hq/pull/24419/commits/7f148d85647b7cb465bb143d52950f6f16035f9f

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 2 above)
Comment (millerdev) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: Good idea. I'll do that in a separate PR.

Comment (millerdev) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: https://github.com/dimagi/commcare-hq/pull/24431

**Pull request merged**